### PR TITLE
Add assembly validation example

### LINKED ISSUE #38962: ApiCompatValidateAssemblies docs don't mention how to specify the baseline assemblies
**Issue description**

I was reading the https://learn.microsoft.com/en-us/dotnet/fundamentals/apicompat/assembly-validation document and see that it tells folks how to turn on the task, but not how to provide inputs.  It does mention:
> Validates that there are no breaking changes across versions.

I think it will get it's multi-targeting inputs from the project automatically, but it needs the author to tell it how to find the previous version to compare to.

I think the way to do this is to specify `ApiCompatContractAssembly` and do so in a way that's specific to each TargetFramework of the project.

It would be nice to see a sample for how someone might do this.

cc @ViktorHofer @gewarren 

---
[Associated WorkItem - 198657](https://dev.azure.com/msft-skilling/Content/_workitems/edit/198657)

### PR DESCRIPTION
**Pull request by gewarren**
Fixes #38962.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/fundamentals/apicompat/assembly-validation.md](https://github.com/dotnet/docs/blob/848cd488dda6a12199acd023a2bcb1b017da2bc5/docs/fundamentals/apicompat/assembly-validation.md) | [Assembly validation](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/apicompat/assembly-validation?branch=pr-en-us-39764) |
| [docs/fundamentals/apicompat/overview.md](https://github.com/dotnet/docs/blob/848cd488dda6a12199acd023a2bcb1b017da2bc5/docs/fundamentals/apicompat/overview.md) | [API compatibility tools](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/apicompat/overview?branch=pr-en-us-39764) |

### CODE REVIEW

File: docs/fundamentals/apicompat/assembly-validation.md [hunk 1]
```diff
@@ -19,14 +19,15 @@ You can run assembly validation either as an [MSBuild task](#enable-msbuild-task
...
+You enable assembly validation in your .NET project by setting the [`ApiCompatValidateAssemblies`](../../core/project-sdk/msbuild-props.md#apicompatvalidateassemblies) property to `true` and specifying the path to the *contract* (baseline) assembly. You must also add a package reference to [Microsoft.DotNet.ApiCompat.Task](https://www.nuget.org/packages/Microsoft.DotNet.ApiCompat.Task). (The `targets` files in that package aren't part of the .NET SDK.)
 
 ```xml
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>netstandard2.0;net6.0</TargetFrameworks>
+    <TargetFrameworks>net8.0</TargetFrameworks>
     <ApiCompatValidateAssemblies>true</ApiCompatValidateAssemblies>
+    <ApiCompatContractAssembly>[Path to contract assembly]</ApiCompatContractAssembly>
```
Comment (gewarren) [inline docs/fundamentals/apicompat/assembly-validation.md:30]: @ViktorHofer How do you specify the path(s) if you are multi-targeting?

File: docs/fundamentals/apicompat/assembly-validation.md [hunk 2]
```diff
@@ -39,6 +40,50 @@ You enable assembly validation in your .NET project by setting the [`ApiCompatVa
...
+
+   ```csharp
+   namespace ValidateMe
+   {
+       public interface IAnimal
+       {
+           string Name { get; }
+           //string Sound { get; }
+       }
+   }
```
Comment (BillWagner) [inline docs/fundamentals/apicompat/assembly-validation.md:55]: I'd make a file scoped namespace here:

```suggestion
   namespace ValidateMe;
   
   public interface IAnimal
   {
       string Name { get; }
       //string Sound { get; }
   }
```

**Review by BillWagner — COMMENTED** [review]
This looks great @gewarren 

I just had one small suggestion.

**Comment by gewarren** [comment]
@BillWagner Could you approve this one?

**Review by BillWagner — APPROVED** [review]

**Pull request merged**